### PR TITLE
Add info.cemu.Cemu

### PR DESCRIPTION
**Pull request by ColinKinloch**
### Please confirm your submission meets all the criteria

- [x] I have read the [App Requirements][reqs] and [App Maintenance][maint] pages.
- [x] My pull request follows the instructions at [App Submission][submission].
- [x] I am using only the minimal set of permissions. *(If not, please explain each non-standard permission.)*
- [x] All assets referenced in the manifest are redistributable by any party.  If not, the unredistributable parts are using an extra-data source type.
- [x] I am an upstream contributor to the project. If not, I contacted upstream developers about submitting their software to Flathub. **Link:** https://github.com/flathub/flathub/pull/3975#issuecomment-1480245190
- [ ] I own the domain used in the application ID or the domain has a policy for delegating subdomains (e.g. GitHub, SourceForge).
- [x] Any additional patches or files have been submitted to the upstream projects concerned. *(If not, explain why.)* 

[reqs]: https://github.com/flathub/flathub/wiki/App-Requirements
[maint]: https://github.com/flathub/flathub/wiki/App-Maintenance
[submission]: https://github.com/flathub/flathub/wiki/App-Submission

### To Do

- [x] Filesystem permissions
- [x] Discord integration
- [x] Upstream `-maes` patch https://github.com/cemu-project/Cemu/pull/696
- [x] Testing
  + [ ] ~~Discord integration~~ (failed :shrug: )

### CODE REVIEW

File: info.cemu.Cemu.yml [hunk 1]
```diff
@@ -0,0 +1,185 @@
...
+          url-template: https://github.com/fmtlib/fmt/archive/$version.tar.gz
+
+  - name: wxwidgets
+    buildsystem: autotools
+    config-opts:
+      - --with-gtk=3
+      - --disable-sdltest
+      - --enable-shared
+      - --enable-webview
+      - --enable-webviewwebkit
```
Comment (hfiguiere) [inline info.cemu.Cemu.yml:150]: are you sure these works? AFAIK this require GtkWebKit which is definitely not in the KDE runtime.

File: info.cemu.Cemu.yml [hunk 2]
```diff
@@ -0,0 +1,185 @@
+app-id: info.cemu.Cemu
+runtime: org.kde.Platform
```
Comment (hfiguiere) [inline info.cemu.Cemu.yml:2]: I don't see where this uses Qt for UI.

**Comment by ColinKinloch** [comment]
bot, build info.cemu.Cemu

File: info.cemu.Cemu.yml [hunk 3]
```diff
@@ -0,0 +1,189 @@
...
+        sha256: 5dea48d1fcddc3ec571ce2058e13910a0d4a6bab4cc09a809d8b1dd1c88ae6f2
+        x-checker-data:
+          type: anitya
+          project-id: 11526
+          url-template: https://github.com/fmtlib/fmt/archive/$version.tar.gz
+
+  - name: wxwidgets
+    buildsystem: autotools
+    config-opts:
+      - --with-gtk=3
```
Comment (hfiguiere) [inline info.cemu.Cemu.yml:152]: note that you removed the webview here. If all you need is gtk3, then freedesktop.sdk should be enough.

**Comment by ColinKinloch** [comment]
I had a look at what's included in the freedesktop platform.
Is building `glslang` necessary?
When I try to build Cemu it without I get:
```
CMake Error at /usr/share/glslang/glslang-targets.cmake:130 (message):
  The imported target "glslang::MachineIndependent" references the file

     "/usr/lib/x86_64-linux-gnu/libMachineIndependent.a"

  but this file does not exist.  Possible reasons include:

  * The file was deleted, renamed, or moved to another location.

  * An install or uninstall procedure did not complete successfully.

  * The installation package was faulty and contained

     "/usr/share/glslang/glslang-targets.cmake"

  but not all the files it references.

Call Stack (most recent call first):
  /usr/share/glslang/glslang-config.cmake:26 (include)
  CMakeLists.txt:106 (find_package)
```

**Comment by barthalion** [comment]
Sounds so, if it fails!

bot, build info.cemu.Cemu

**Comment by barthalion** [comment]
Can you contact upstream about this submission, unless you already did?

**Comment by ColinKinloch** [comment]
@Exzap, would you be okay with Cemu being distributed via flathub? Are there any problems you can see with this build?

**Comment by ColinKinloch** [comment]
The AES patch is still applied because the build isn't targeting the `main` branch, it's currently the `v2.0-29` branch.
As for testing I've been playing Zelda Twilight Princess HD with motion controls, I have noticed one issue with the GamePad view under wayland:
[Screencast from 2023-03-22 00-54-04.webm](https://user-images.githubusercontent.com/334272/226774659-075da073-f69b-44cc-b7a6-5af5b0427f21.webm)
Which I suppose could be remedied by removing the wayland socket permission.

Is there any specific test you'd suggest?

**Comment by Exzap** [comment]
> The AES patch is still applied because the build isn't targeting the main branch, it's currently the v2.0-29 branch.

Oh right, I forgot that for flatpak you have to use a specific tag. 2.0-29 is actually pretty stable so we should stick to that one for now until the next stable release is ready.

> Is there any specific test you'd suggest?

Nah was just curious if this plays any games at all or if its a case of "builds, but doesn't work yet". If there are smaller issues (like the flickering GamePad view) we can fix them as we go.

Anyway, this looks good to me. @ColinKinloch has my approval.

**Comment by ColinKinloch** [comment]
Would adding `--filesystem=xdg-data/Cemu:create` make sense?
It would allow Cemu to store save files in `~/.local/share/Cemu`.
I haven't seen many other flathub apps use it.

**Comment by JakobDev** [comment]
Flatpaks Apps should store their data in `~/.var/apps/<id>` and not in another directory. That's part of the Sandboxing and would cause problems if not.

**Comment by ColinKinloch** [comment]
Cemu uses `wxFileDialog` and `wxDirDialog`. Unless I'm missing something wxwidgets doesn't support portals so should I add `--filesystem=host:ro` to allow users to set their game directories?

**Comment by ColinKinloch** [comment]
I have, it doesn't, I've been specifying `--filesystem=/mnt/data/WiiUGames` whole testing. I was more wondering if `host:ro` is the go to in this situation or if there's some build flag I'm missing from wx.

**Comment by ColinKinloch** [comment]
bot, build info.cemu.Cemu

**Comment by ColinKinloch** [comment]
I've set up discord rich presence based on other flatpaks.
I've not gotten it working with either this or any other flatpak.
I suppose I'll leave it to someone who uses that feature to test / fix it.

**Comment by barthalion** [comment]
/merge @Exzap

@Exzap, when the app is up on flathub.org in a few hours, can you please log in at https://beta.flathub.org/login and verify ownership of the app? It will add a verification badge on the website and beta.flathub.org will be made "stable" soon.

**Pull request closed without merging**